src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java

```diff
@@ -1681,8 +1682,15 @@ && ofNullable(caseData.getRespondent2Copy()).isPresent()) {
             updatedData.respondent2DetailsForClaimDetailsTab(updatedRespondent2);
         }
 
-        CaseRole respondentTwoCaseRoleToCheck;
+        if (caseData.getDefenceAdmitPartPaymentTimeRouteRequired() != null
+            &&  caseData.getDefenceAdmitPartPaymentTimeRouteRequired().equals(IMMEDIATELY)) {
```

Comment:
```suggestion
            &&  caseData.getDefenceAdmitPartPaymentTimeRouteRequired() == IMMEDIATELY) {
```

Double Equal Sign (“==”) is null-safe whereas the equals() method is not. The equals() is a method of Enum class. If we try to call that method on a NULL reference we get a NullPointerException . 

“==” provides null-safety and type-safety during compile-time.